src/server.mjs

```diff
@@ -1154,13 +1322,31 @@ function getBedRegions(bedFile, dataPath) {
     }
     bed_info["chunk"].push(chunk);
 
+  });
 
 
+  // check for a tracks.json file to prefill tracks configuration
+  for (let i = 0; i < bed_info["chunk"].length; i++) {
     let tracks = {};
     let trackID = 1;
 
-    const chunk_path = `${dataPath}${chunk}`;
-    const track_json = path.join(chunk_path, "tracks.json");
+    const dirName = bed_info["chunk"][i];
+    const chunk_path = `${dataPath}${dirName}`;
+    let track_json = `${chunk_path}/tracks.json`
+    
+    // download the tracks file if bed is an url
+    if (isURL) {
+      // 1 level above the bedurl
+      const baseURL = bedFile.split('/').slice(0, -1).join('/') + '/';
+      await retrieveChunk(baseURL, dirName, false);
+    }
+
+    const tempTracksPath = path.resolve(config.tempDirPath, sanitize(dirName), "tracks.json");
+    // check the temporary directory for downloaded tracks info
+    if (fs.existsSync(tempTracksPath)) {
+      track_json = tempTracksPath;
+    }
```

Comment:
What if this file happens to exist, but the BED file didn't come from a URL and was actually a local file?